dockerfile_parse/parser.py

```diff
@@ -100,6 +107,15 @@ def __init__(self, path=None, cache_content=False, env_replace=True, parent_env=
         else:
             self.parent_env = {}
 
+    @contextmanager
+    def _open_dockerfile(self, mode):
+        if self.fileobj is not None:
+            self.fileobj.seek(0)
```

Comment:
I think that looks right, if that's how we want to handle it.

I just wonder if that's the correct approach. For mode=='r' it seems fine, but for mode=='w' should we fail? We know we've already read from `fileobj` in that case, so we know it's not at the right position.